ocis/pkg/runtime/service/service.go

```diff
@@ -519,6 +523,8 @@ func trapShutdownCtx(s *Service, srv *http.Server, ctx context.Context) {
 		s.Log.Info().Msg("tcp listener shutdown")
 	}()
 
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
better move the unlock after the `for` loop, otherwise we'll need to wait until all the services have shutdown of we reach the `_defaultInterruptTimeoutDuration` timeout; it will take too much time to unlock the mutex and it could block other goroutines for a long time.